src/uucore/src/lib/features/entries.rs

```diff
@@ -83,13 +83,14 @@ pub fn get_groups() -> IOResult<Vec<gid_t>> {
         if res == -1 {
             let err = IOError::last_os_error();
             if err.raw_os_error() == Some(libc::EINVAL) {
-                // Number of groups changed, retry
+                // Number of groups has increased, retry
```

Comment:
are you sure it is always increased ?

Reply:
POSIX.1-1988 on `getgroups` return value:

```
[EINVAL]
The gidsetsize argument is non-zero and less than the number of group IDs that would have been returned.
```

Since `ngroups` is based on earlier `getgroups` return value, the only case where `EINVAL` is returned when the indicated size (`ngroups` parameter for `getgroups`) is less than what the return list would be. If the reserved space is larger than the actual list, then `res != -1`.

Reply:
oh, cool, thanks :)